jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -56,38 +60,41 @@
         requiresDependencyCollection = ResolutionScope.COMPILE)
 public class SchemaGeneratorMojo extends AbstractMojo {
 
-    @Parameter(defaultValue = "${project}", required = true, readonly = true)
-    MavenProject project;
+    /**
+     * Full name of the classes for which the JSON schema will be generated.
+     */
+    @Parameter(property = "classNames")
+    private String[] classNames;
 
     /**
-     * Full name of the class for which the JSON schema will be generated.
+     * Full name of a package for which for all classes in the package JSON schemas will be generated.
      */
-    @Parameter(property = "className", required = true)
-    private String className;
+    @Parameter(property = "packageNames")
+    private String[] packageNames;
 
     /**
      * The directory path where the schema files are generated.
      * <br>
      * By default, this is: {@code src/main/resources}
      */
-    @Parameter(property = "schemaFilePath")
+    @Parameter
     private File schemaFilePath;
 
     /**
      * The name of the file in which the generated schema is written. Allowing for two placeholders:
      * <ul>
-     * <li><code>{0}</code> - containing the simple class name (last part of the {@link #className} parameter)</li>
-     * <li><code>{1}</code> - containing the package path (first part of the {@link #className} parameter)</li>
+     * <li><code>{0}</code> - containing the simple class name (last part of the className)</li>
+     * <li><code>{1}</code> - containing the package path (first part of the className)</li>
      * </ul>
      * The default name is: <code>{0}-schema.json</code>
      */
-    @Parameter(property = "schemaFileName", defaultValue = "{0}-schema.json")
+    @Parameter(defaultValue = "{0}-schema.json")
```

Comment:
As per comment on the `schemaFilePath`, I don't see a reason to prevent this parameter to be available via the command-line.
```suggestion
    @Parameter(property = "schemaFileName", defaultValue = "{0}-schema.json")
```

Reply:
Agree. My bad.
Fixed.